### DISCUSSION TITLE
Split code history for standard and debugging consoles?

### DISCUSSION THREAD

**Discussion by etiennebacher** [posit-dev/positron#4841] [Feedback] [answered]
Hello, first time trying Positron, I really like the experience so far!

There's one thing I miss from RStudio regarding the use of `browser()`. In RStudio, when I navigate to previous code in the console using the up and bottom arrows, the code history of the "standard console" and the one of the "debugging console" are separated. 

For example, suppose I want to debug a function `foo()`: I put a `browser()` call in it and then call `foo()` in the console, which triggers the debugging console. Inside it, I can define new variables or print the value of existing variables. When I exit the debugging console, if I go up with the arrow to see the previous code, the previous command is `foo()`, it doesn't remember what I called in the debugging console. Similarly, if I enter the debugging console again by calling `foo()`, going up with the arrow shows the last code called in the previous debugging console, not the call to `foo()`.

Currently, in Positron, the two code histories are mixed, so going up with the arrow to see the previous commands will mix those called in a standard console and those called in a debugging console, which I find less useful (how often do you want to run the code you used in the debugging console once you have exited it?).

Is this feature of RStudio something you would consider implementing in Positron?

Thanks for the great tool!

**Comment by jmcphers** [comment, marked as answer]
Glad you're enjoying the experience and thanks for the feedback! We agree and have had this on the backlog for a while. Here's where we're tracking the work; follow for updates: https://github.com/posit-dev/positron/issues/4478
